src/MediawikiApiInterface.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Mediawiki\Api;
+
+/**
+ * @since 2.2
+ * @licence GNU GPL v2+
+ * @author Addshore
+ * @author Jeroen De Dauw < jeroendedauw@gmail.com >
+ */
+interface MediawikiApiInterface extends ApiRequester, AsyncApiRequester {
+
+	/**
+	 * @since 2.2
+	 *
+	 * @return bool|string false or the name of the current user
+	 */
+	public function isLoggedin();
+
+	/**
+	 * @since 2.2
+	 *
+	 * @param ApiUser $apiUser
+	 *
+	 * @throws UsageException
+	 * @return bool success
+	 */
+	public function login( ApiUser $apiUser );
```

Comment:
I think this method is a design flaw. Now the user of the service needs both the service and the user credentials value object. For most users this is probably annoying (it certainly is for the use case in the WMDE fundraising code).

To get around this you can create some `AutoLoggingInMediawikiApi` decorator that gets the credentials in its constructor and does the login on first post or something like that. However such a dedocrator would still be forced to implement the login method if it wants to implement this interface. It could throw an exception for that method. Still, not nice.

I'm not sure if it's better to omit the method from this interface altogether because of this.